chainer/utils/meta.py

```diff
@@ -0,0 +1,79 @@
+import warnings
+
+import six
+
+
+def final(*args, **kwargs):
+    """Decorator to declare a method final.
+
+    By default, :class:`TypeError` is raised when the decorated method is being
+    overridden.
+
+    The class in which the decorated method is defined must inherit from a
+    base class returned by :meth:`~chainer.utils.enable_final`.
+
+    Args:
+        action: Specifies what happens when the decorated method is being
+            overridden. If `'error'`, :class:`TypeError` is raised. If
+            :class:`DeprecationWarning`, a deprecation warning is raised.
```

Comment:
How about align the type of both values, i.e., either using `TypeError` for the error case or using `'warning'` for the warning case?

Reply:
Fixed by using types.